src/micro_http/src/connection.rs

```diff
@@ -288,7 +292,17 @@ impl<T: Read + Write> HttpConnection<T> {
                 // line end sequence.
                 if *line_start_index == 0 && end_cursor == BUFFER_SIZE {
                     // Header line is longer than BUFFER_SIZE bytes, so it is invalid.
-                    return Err(ConnectionError::ParseError(RequestError::InvalidHeader));
+                    if let Ok(utf8_string) = String::from_utf8(self.buffer.to_vec()) {
+                        return Err(ConnectionError::ParseError(RequestError::InvalidHeader(
+                            InvalidHeaderError::SizeLimitExceeded(utf8_string),
```

Comment:
I do not think `SizeLimitExceeded` is a subcase of `InvalidHeader`. Do you know if HTTP header line/field has maximum 1024 chars, according to the RFC? I was not able to find such limitation and I am inclined to believe that this is just an implementation limitation. I think these limits are configurable for HTTP servers/clients and they are imposed at the level of HTTP requests/response size. If this is the case, we might want to make from `SizeLimitExceeded` a subcase of `RequestError`. Let me know what you think.

Reply:
Synced offline with @AlexandruCihodaru . We've agreed on having `SizeLimitExceeded` as a subcase for an `HttpHeaderError` parent case, with the mention that this needs also a comment for the `SizeLimitExceeded` error, to underline the fact that it is a `micro-http` limitation rather than a hard limit for header fields.

This is similar to the case of parsing request lines, which is also limited to maximum `BUFFER_SIZE`. Here, the result in an `InvalidRequest` error. This can be further specialized in the future to something like `SizeLimitExceeded` in the context of a more generic `RequestLineError`.

Resolving this for now.